wgengine/packet/ip.go

```diff
@@ -70,19 +70,18 @@ type IPHeader struct {
 
 const ipHeaderLength = 20
 
-func (h IPHeader) Length() int {
+func (h IPHeader) Len() int {
 	return ipHeaderLength
 }
 
 func (h IPHeader) Marshal(buf []byte) error {
 	if len(buf) < ipHeaderLength {
 		return errSmallBuffer
 	}
-	length := len(buf)
 
 	buf[0] = 0x40 | (ipHeaderLength >> 2) // IPv4
 	buf[1] = 0x00                         // DHCP, ECN
-	put16(buf[2:4], uint16(length))
+	put16(buf[2:4], uint16(len(buf)))
```

Comment:
you're not going to return an error if len(buf) is too big? This silently wraps around.

Reply:
Fair; I thought it was highly unlikely that we would end up in that situation (packet buffers should be at most `device.MaxContentSize` bytes anyway, which is always fewer than `1 << 16`), but it'd be absurd for me to worry about adding an extra check.